### PR TITLE
Reconcile project structure

### PR DESCRIPTION
**Pull request by mooselumph**
## Why are these changes needed?

Some renaming to get us closer to a consistent project structure. 

- Move Dockerfiles to associated cmd directory
- Make `disperser` directory self-similar with broader repo
  - Rename `proto` to `api`
  - Create `common` folder for shared components like blobstore
  - Move `dispatcher` and `confirmer` to `batcher` directory. 

This makes the `disperser` folder easier to read and understand.

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: docker-compose-build.yaml [hunk 1]
```diff
@@ -3,22 +3,22 @@ services:
...
       context: .
-      dockerfile: disperser/batcher.Dockerfile
+      dockerfile: disperser/cmd/batcher/Dockerfile
     image: ghcr.io/layr-labs/eigenda/batcher:${BUILD_TAG:-latest}
   disperser:
     build:
       context: .
-      dockerfile: disperser/disperser.Dockerfile
+      dockerfile: disperser/cmd/apiserver/Dockerfile
     image: ghcr.io/layr-labs/eigenda/disperser:${BUILD_TAG:-latest}
```
Comment (dmanc) [inline docker-compose-build.yaml:12]: should we rename the image to apiserver?

Comment (mooselumph) [reply docker-compose-build.yaml:12]: I think this would make sense.

**Review by jianoaix — APPROVED** [review]
LG!

File: docker-compose-build.yaml (hunk 1 above)
Comment (mooselumph) [reply docker-compose-build.yaml:12]: Will do this in a separate PR since it requires coordination with devops repo.

**Pull request merged**